src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorDiagnosticsPublisher.cs

```diff
@@ -195,10 +195,19 @@ internal async Task PublishDiagnosticsAsync(IDocumentSnapshot document)
             TextDocument = new TextDocumentIdentifier { Uri = uriBuilder.Uri },
         };
 
-        var delegatedResponse = await _languageServer.SendRequestAsync<DocumentDiagnosticParams, SumType<FullDocumentDiagnosticReport, UnchangedDocumentDiagnosticReport>?>(
-            RazorLanguageServerCustomMessageTargets.RazorPullDiagnosticEndpointName,
-            delegatedParams,
-            CancellationToken.None).ConfigureAwait(false);
+        SumType<FullDocumentDiagnosticReport, UnchangedDocumentDiagnosticReport>? delegatedResponse = null;
+        try
+        {
+            // TODO: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1806769
+            delegatedResponse = await _languageServer.SendRequestAsync<DocumentDiagnosticParams, SumType<FullDocumentDiagnosticReport, UnchangedDocumentDiagnosticReport>?>(
+                RazorLanguageServerCustomMessageTargets.RazorPullDiagnosticEndpointName,
+                delegatedParams,
+                CancellationToken.None).ConfigureAwait(false);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Failed to get C# diagnostics");
```

Comment:
Nit: This isn't just C# diagnostics, this is diagnostics from any delegated server.